src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -29,6 +31,82 @@ constexpr auto DELAYED_EVENTS_QUEUE_PATH = "queue/vd/delayed";
 constexpr auto SLEEP_RETRY_THREADS_SEC = 60;
 
 int SOCKET_WAIT = 0;
+constexpr auto COMPRESSED_DB_PATH {"vd_1.0.0_vd_4.8.0.tar.xz"};
+constexpr auto DECOMPRESSED_DB_PATH {"vd_1.0.0_vd_4.8.0.tar"};
+constexpr auto VD_STATE_QUEUE_PATH = "queue/vd/state_track";
+constexpr auto VD_KEYSTORE_PATH = "queue/keystore";
+constexpr auto VD_DATABASE_PATH {"queue/vd"};
+
+bool VulnerabilityScannerFacade::decompressDatabase(const std::string& databaseVersion)
+{
+    bool ret = false;
+
+    // Check database version and presence. It will attempt to decompress the database
+    // if the version doesn't not match, the state_track does not have information
+    // or there's no VD DB.
+    if ((databaseVersion.compare(__ossec_version) != 0 || databaseVersion.empty() ||
+         !std::filesystem::exists(VD_DATABASE_PATH)) &&
+        std::filesystem::exists(TMP_DIR))
+    {
+        // Search in path for a compressed database file.
+        std::regex regexPattern(VD_COMPRESSED_FILE_PATTERN);
+        std::string compressedFilePath;
+        size_t compressedFilesNumber = 0;
+        for (const auto& item : std::filesystem::directory_iterator(TMP_DIR))
+        {
+            if (std::regex_match(item.path().filename().string(), regexPattern))
+            {
+                compressedFilePath = item.path().string();
+                ++compressedFilesNumber;
+            }
+            if (compressedFilesNumber > 1)
+            {
+                throw std::runtime_error("Multiple database compressed files detected");
+            }
```

Comment:
Checking again the sequence diagram, we shouldn't throw an exception. The process must continue 

```suggestion
                logDebug2(WM_VULNSCAN_LOGTAG, "Found compressed database file: %s", item.path().string());
                compressedFilePath = item.path().string();
                break;
            }
```

Reply:
This change is doing the same before the check to of multiple tar.xz files.